frontend/src/services/blockServices.js

```diff
@@ -0,0 +1,90 @@
+export const getAllBlocks = async () => {
+  const URL = `${import.meta.env.VITE_BACKEND_URL}/api/v1/blocks`;
+
+  try {
+    const res = await fetch(URL);
+    const data = await res.json();
+
+    if (!res.ok) {
+      throw new Error(data.message || 'Failed to fetch blocks');
+    }
+
+    return data.payload;
+  } catch (error) {
+    throw new Error(error.message || 'Failed to fetch blocks');
+  }
+};
+
+export const getBlock = async () => {
```

Comment:
This looks the same as  ` line 1 to 16 `, if it is, you can remove ` line 18 to 33`